From 285bde170ffeaaa0ec7da11cbe51ef7c675b2c57 Mon Sep 17 00:00:00 2001
From: "Carsten Haitzler (Rasterman)" <raster@rasterman.com>
Date: Sat, 22 May 2021 14:19:19 +0100
Subject: [PATCH 08/13] alpm - fix warning of casing ptr to gboolean - this is
 problematic

a ptr canbe 64bits. casting to a gboolean is lossy and if the ptr just
happens to be the right value - we can accidentally make this if
false. just use pointer as-is thus only 0/null will be false as intended
---
 backends/alpm/pk-alpm-sync.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backends/alpm/pk-alpm-sync.c b/backends/alpm/pk-alpm-sync.c
index 5ad94245c..747b263f5 100644
--- a/backends/alpm/pk-alpm-sync.c
+++ b/backends/alpm/pk-alpm-sync.c
@@ -191,7 +191,7 @@ pk_backend_sync_thread (PkBackendJob* job, GVariant* params, gpointer p)
 	if (!only_trusted && !pk_alpm_disable_signatures (backend, &error))
 		goto out;
 
-	if ((gblooean)p) {
+	if (p) {
 		i = alpm_get_syncdbs(priv->alpm);
 		pk_alpm_refresh_databases (job, TRUE, i, &error);
 	}
-- 
2.31.1

